### PR TITLE
[clangd] Add clangd 18 release notes

### PR DESCRIPTION
**Pull request by HighCommander4**

### CODE REVIEW

**Comment by HighCommander4** [comment]
To prepare these, I went through the clangd commits in clangd 18 and made note of what seemed notable to me.

In some areas of the code that I'm less familiar with, I was somewhat vague (e.g. under "Diagnostics" I added "Improved quality of include-cleaner diagnostics (missing and unused includes)"; if someone more familiar with include-cleaner would like to expand on that with more specific improvements, that would be welcome).

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -51,21 +51,40 @@ Improvements to clangd
...
+    * Parameter hints are now shown for calls through function pointers
+    * Parameter hints are now shown for calls to a class's ``operator()``
+    * No longer show bogus parameter hints for some builtins like ``__builtin_dump_struct``
+
 Diagnostics
 ^^^^^^^^^^^
 
-Semantic Highlighting
-^^^^^^^^^^^^^^^^^^^^^
+- Improved quality of include-cleaner diagnostics (missing and unused includes)
```
Comment (kadircet) [inline clang-tools-extra/docs/ReleaseNotes.rst:65]: i'd move this one into include-cleaner release notes (I am also happy to do that myself), a more detailed version could look like:

- Support for `--only-headers` flag to limit analysis to headers matching a regex
- Recognizes references through `concept`s
- Builtin headers are not analyzed
- Handling of references through `friend` declarations
- Fixes around handling of IWYU pragmas on stdlib headers
- Improved handling around references to/from template specializations

**Review by kadircet — APPROVED** [review]
thanks a lot for doing this @HighCommander4 !

File: clang-tools-extra/docs/ReleaseNotes.rst (hunk 1 above)
Comment (HighCommander4) [reply clang-tools-extra/docs/ReleaseNotes.rst]: Thanks, I added these in an include-cleaner section

**Comment by HighCommander4** [comment]
@kadircet do you know what is the process for getting this approved and merged? As this is targeting the `release/18.x` branch, I do not have permissions to merge it myself like a regular PR.

**Comment by kadircet** [comment]
i believe @tstellar should have that power :rocket: 

Do you mind merging this into the branch, or advise on how we should do that instead?

**Pull request merged**